src/App.svelte

```diff
@@ -31,8 +32,16 @@
     connectionDialogState,
   } from './script/stores/connectDialogStore';
   import { isLoading } from 'svelte-i18n';
+  import { fetchCachedBrowserInfo } from './utils/api';
+  let isPotentiallyNextGenUser: boolean = false;
 
   onMount(() => {
+    (async () => {
+      const { country } = await fetchCachedBrowserInfo($consent);
+      // Show redirect dialog if user's location is UK or Jersey
+      isPotentiallyNextGenUser = country === 'GB' || country === 'JE';
```

Comment:
Started an internal discussion about the detail of this.

Reply:
Thanks, I have updated the UI and the country detection part